src/components/Barrier.jsx

```diff
@@ -47,16 +48,18 @@ export default connect(
         belowShade: store.belowShade.clone(),
         betweenShade: store.betweenShade.clone(),
         barrierColor: store.barrierColor,
+        hidePriceLines: store.hidePriceLines,
     }),
     (store, {
-        color, shade, high, low, relative, draggable, onBarrierChange,
+        color, shade, high, low, relative, draggable, onBarrierChange, hidePriceLines,
     }) => {
         if (color) { store.barrierColor = color; }
         if (shade) { store.shadeState = `SHADE_${shade}`.toUpperCase(); }
-        if (high) { store.high_barrier = high; }
-        if (low) { store.low_barrier = low; }
-        if (relative) { store.relative = relative; }
-        if (draggable) { store.draggable = draggable; }
+        if (high !== undefined) { store.high_barrier = high; }
+        if (low !== undefined) { store.low_barrier = low; }
+        if (relative !== undefined) { store.relative = relative; }
+        if (draggable !== undefined) { store.draggable = draggable; }
         if (onBarrierChange) { store.onBarrierChange = onBarrierChange; }
+        store.setHidePriceLines(hidePriceLines);
```

Comment:
@brucebinary use `store.hidePriceLines = hidePriceLines;` instead.

The callback the receives `(store, { color, shade ... highPriceLines}) => { }` is already wrapped inside a mobx action in the `connect()` method.

Reply:
Interesting...

done.